CMakeLists.txt

```diff
@@ -41,8 +41,6 @@ set(AER_SIMULATOR_CPP_EXTERNAL_LIBS
 	"${AER_SIMULATOR_CPP_SRC_DIR}/third-party/win64/lib"
 	"${USER_LIB_PATH}")
 
-set(AER_COMPILER_DEFINITIONS "")
```

Comment:
This was meant to be a placeholder for all Aer compiler definitions, not only Thrust related ones. Every new compiler definition should be added here.
We don't really want to pollute the space with specific variables for every use case, we can end up with dozens of variables which is harder to understand and maintain.
I'm suggesting to stick with AER_COMPILER_DEFINITIONS variable, and add all Thrust required definitions in here, instead of creating AER_THRUST_SUPPORTED_DEFINITION / AER_THRUST_DEVICE_SYSTEM_DEFINITION

Reply:
The current cmake generates wrong parameters.
```
$ cmake -version
cmake version 3.14.5
$ cat CMakeFiles/qasm_simulator.dir/flags.make | grep CXX_DEFINES
CXX_DEFINES = -D THRUST_DEVICE_SYSTEM="THRUST_DEVICE_SYSTEM_OMP AER_THRUST_SUPPORTED"
```
 I guess that use of `target_compile_definitions` is wrong.

Reply:
Oh Ok, then maybe we have to use ";" or ":"  for splitting between variables... let me take a look.

Reply:
Ok, we only need to remove the quotes, so `AER_COMPILER_DEFINITIONS` is now a list instead of a string:
Change this:
```
set(AER_COMPILER_DEFINITIONS "${AER_COMPILER_DEFINITIONS} THRUST_DEVICE_SYSTEM=THRUST_DEVICE_SYSTEM_OMP")
```
to this:
```
set(AER_COMPILER_DEFINITIONS ${AER_COMPILER_DEFINITIONS} THRUST_DEVICE_SYSTEM=THRUST_DEVICE_SYSTEM_OMP)
```
... in all the places.

Reply:
Another option would be using CMake `list()` function like:
```
list(APPEND AER_COMPILER_DEFINITIONS THRUST_DEVICE_SYSTEM=THRUST_DEVICE_SYSTEM_OMP)
```
Whatever you like the most... I really don't mind